GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -402,8 +403,19 @@ private bool TryGetSymLinkTarget(string sha, out string symLinkTarget)
                                 if (bufferIndex == buffer.Length)
                                 {
                                     bufferIndex = 0;
+                                    bufferSize += buffer.Length;
                                 }
                             }
+                            bufferSize += Convert.ToInt32(bufferIndex);
+
+                            if (bufferSize != blobLength)
+                            {
+                                // If the read size does not match the expected size print an error and add the file to ModifiedPaths.dat
+                                // This allows the user to see that something went wrong with file hydration
+                                // Unfortunitely we must do this check *after* the file is hydrated since the header isn't corrupt for trunctated objects on mac
+                                this.OnLogError($"Read {relativePath} to {bufferSize}, not expected size of {blobLength}");
+                                this.FileSystemCallbacks.OnFileConvertedToFull(relativePath);

```

Comment:
The end result is the same, but I think the intent of this call might be clearer if we call `OnFailedPlaceholderUpdate` instead.

Another option would be to add a new `OnFailedFileHydration` background task that does the same thing.

The only *potential* functionality impact would be if one day we decide to change what the BG thread does for these tasks.  If a user were to upgrade a new version of VFS4G with pending BG tasks, the persisted `OnFileConvertedToFull` or `OnFailedPlaceholderUpdate` tasks that we scheduled here could be processed differently.

cc: @kewillford for his thoughts, currently we only call `OnFailedPlaceholderUpdate` when we fail to update a placeholder as part of a projection change.

Reply:
I added `OnFailedFileHydration `, I think there is value in being explicit here.  As @wilbaker points out it also protects us if changes are made in the future to other tasks.

Reply:
> I added OnFailedFileHydration 

This call to `OnFileConvertedToFull` still needs to be updated (to `OnFailedFileHydration`).